src/main/java/org/junitpioneer/jupiter/cartesian/CartesianTestExtension.java

```diff
@@ -90,39 +107,58 @@ private static void ensureNoInputConflicts(List<?> methodSources, List<?> parame
 
 	private List<?> getSetFromAnnotation(ExtensionContext context, Annotation source, Parameter parameter) {
 		try {
-			CartesianArgumentsProvider<?> provider = initializeArgumentsProvider(source, parameter);
+			CartesianParameterArgumentsProvider<?> provider = initializeParameterArgumentsProvider(source, parameter);
 			return provideArguments(context, parameter, provider);
 		}
 		catch (Exception ex) {
 			throw new ExtensionConfigurationException("Could not provide arguments because of exception.", ex);
 		}
 	}
 
-	private CartesianArgumentsProvider<?> initializeArgumentsProvider(Annotation source, Parameter parameter) {
+	private CartesianMethodArgumentsProvider initializeMethodArgumentsProvider(Annotation source, Method method) {
+		CartesianArgumentsSource providerAnnotation = AnnotationSupport
+				.findAnnotation(method, CartesianArgumentsSource.class)
+				// never happens, we already know these annotations are annotated with @ArgumentsSource
+				.orElseThrow(() -> new PreconditionViolationException(format(
```

Comment:
Shouldn't this then be an `IllegalStateException`? I understand `PreconditionViolationException` to mean that the user did something wrong.